### PR TITLE
[SPARK-8820][Streaming] Add a configuration to set checkpoint dir.

### PR DESCRIPTION
**Pull request by SaintBacchus**
Add a configuration to set checkpoint directory  for convenience to user.
[Jira Address](https://issues.apache.org/jira/browse/SPARK-8820)

### CODE REVIEW

**Comment by srowen** [comment]
I understand the motivation, but I would oppose adding yet more config and API surface area unless there's a clear need. I don't see that this does much.

**Comment by SaintBacchus** [comment]
> adding yet more config and API surface area unless there's a clear need

@srowen  Do you mean it's not necessary to add this config?  
I purpose the user may not hard code the `checkpoint directory` or they must implement the config themself.

**Comment by andrewor14** [comment]
@tdas

**Comment by SaintBacchus** [comment]
@harishreedharan Can you also review this PR plz

**Comment by tdas** [comment]
Please added unit tests in StreamingContextSuite

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 1]
```diff
@@ -204,6 +204,8 @@ class StreamingContext private[streaming] (
 
   private var shutdownHookRef: AnyRef = _
 
+  conf.getOption("spark.streaming.checkpointDir").foreach(checkpoint)
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: Can you rename this configuration to `spark.streaming.checkpoint.directory`. This is to maintain cleanliness in the future if we add more configuration related to checkpointing (`spark.streaming.checkpoint.*`)

**Comment by harishreedharan** [comment]
LGTM

**Pull request closed without merging**